test/hotspot/jtreg/serviceability/dcmd/compiler/PerfMapTest.java

```diff
@@ -113,4 +113,17 @@ public void specifiedDefaultMapFile() throws IOException {
         run(new JMXExecutor(), "Compiler.perfmap " + path.toString(), path);
         Files.deleteIfExists(path);
     }
+
+    @Test
+    public void logErrorsDcmdOutputStream() throws IOException {
+        String test_dir = System.getProperty("test.dir", ".");
+        Path path = Paths.get("nonexistent", test_dir);
+        try {
+            OutputAnalyzer output = new JMXExecutor().execute("Compiler.perfmap %s".formatted(path));
+            output.shouldContain("Failed to create nonexistent/%s for perf map".formatted(test_dir));
+            output.shouldNotHaveExitValue(0);
```

Comment:
Hi Kevin, 

Yes, I noticed the test was exiting with a non-zero value when I was testing. After giving it a bit more thought, that check is not the main purpose of the test and I'm not entirely sure why the JMXExecutor behaves that way. I'll just remove the exit value check to avoid confusion.

Reply:
I think it was returning an "exit value" of -1, the test/lib/jdk/test/lib/process/OutputBuffer.java default.

JMXExecutor doesn't set one as there isn't an exit code...   That could be clearer, actually there most be various issues in that area.  But yes just don't check exit code for a JMX Executor, and jcmd would return zero but we don't want to embed that as a requirement, it's really a failure.

Reply:
> I think it was returning an "exit value" of -1, the test/lib/jdk/test/lib/process/OutputBuffer.java default.

Correct, that was the exit value. 

> But yes just don't check exit code for a JMX Executor, and jcmd would return zero but we don't want to embed that as a requirement, it's really a failure.

Agreed. I removed the exit status check.